src/Module.hs

```diff
@@ -192,6 +192,7 @@ data Decl ty scheme decl expr
   | DefineMacros [(Ident, MacroVar, expr)]
   | Meta decl
   | Example SrcLoc scheme expr
+  | Run expr
```

Comment:
Perhaps this should also take a `SrcLoc` so we can say what failed if there is a failure?

Reply:
Given that this is a private implementation detail rather than a public interface to a library, I'd rather add that kind of thing as it becomes necessary (that is, as part of improving error reporting) rather than now, where it doesn't get used.

Reply:
There are a few places in this PR where it _could_ get used pretty easily - like in the "Impossible - type checker bug" error message. Shouldn't hold up merge, but IMO it would be nice to report a source location in messages like that.

Reply:
Yeah, that would be nice :)

Reply:
As always, your feedback has led to an improvement :-)